File: diagram-poc/frontend/src/app/core/services/recommendation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { apiBaseUrl } from '../app-config';

export interface RecommendationItem {
  type: string; // template | part | solution
  title: string;
  detail: string;
  source: string;
  verify: string;
}

export interface RecommendationResult {
  items: RecommendationItem[];
  model: string;
  aiGenerated: boolean;
  note?: string;
}

const API = apiBaseUrl();

/**
 * AI recommendations (Claude) with source traceability + verify prompts.
 * Falls back to a rule-based engine server-side when no API key is configured.
 */
@Injectable({ providedIn: 'root' })
export class RecommendationService {
  constructor(private http: HttpClient) {}

  recommend(goal: string, currentParts: string[]): Observable<RecommendationResult> {
    return this.http.post<RecommendationResult>(`${API}/recommendations`, { goal, currentParts });
  }
}

File: diagram-poc/backend/src/main/java/com/example/diagram/service/RecommendationService.java
package com.example.diagram.service;

import com.example.diagram.web.dto.RecommendationRequest;
import com.example.diagram.web.dto.RecommendationResult;

/**
 * Recommends templates, parts and solution options with source traceability and
 * prompts to verify specs/datasheets. Backed by Claude when an API key is
 * configured, and by a deterministic rule-based engine otherwise.
 */
public interface RecommendationService {
    RecommendationResult recommend(RecommendationRequest request);
}
